### PR TITLE
Slash timeout archival

### PR DESCRIPTION
**Pull request by rem1niscence**
some nodes on archival are failing due to misconfigured timeouts on their end, this slashes them and removes from the session if fail continously

### CODE REVIEW

File: src/utils/constants.ts [hunk 1]
```diff
@@ -9,3 +9,12 @@ export const WS_ONLY_METHODS = [
   'eth_subscribe',
   'newBlockFilter',
 ]
+
+export const ARCHIVAL_CHAINS = [
```
Comment (rem1niscence) [inline src/utils/constants.ts]: should this be an env?

Comment (nymd) [reply src/utils/constants.ts]: Yes please

File: src/services/cherry-picker.ts [hunk 2]
```diff
@@ -1,9 +1,19 @@
...
 
 const logger = require('../services/logger')
 
+// Amount of times a node is allowed to fail due to misconfigured timeout before
+// being removed from the session
+const TIMEOUT_LIMIT = 20
+
+// Allowed difference on timeout, expressed in seconds, as the timeout usually
+// wont be exact
+const TIMEOUT_THRESHOLD = 2
```
Comment (nymd) [inline src/services/cherry-picker.ts]: maybe TIMEOUT_VARIANCE is a better name for this

File: src/services/cherry-picker.ts [hunk 3]
```diff
@@ -1,9 +1,19 @@
...
 import { Node } from '@pokt-network/pocket-js'
 import { Applications } from '../models'
+import { getNodeNetworkData, removeNodeFromSession } from '../utils/cache'
+import { ARCHIVAL_CHAINS } from '../utils/constants'
 
 const logger = require('../services/logger')
 
+// Amount of times a node is allowed to fail due to misconfigured timeout before
+// being removed from the session
+const TIMEOUT_LIMIT = 20
```
Comment (rem1niscence) [inline src/services/cherry-picker.ts:10]: same for this

File: src/services/cherry-picker.ts [hunk 4]
```diff
@@ -218,6 +235,57 @@ export class CherryPicker {
     await this.redis.set(blockchain + '-' + id + '-service', JSON.stringify(serviceQuality), 'EX', ttl)
   }
 
+  /**
+   * Nodes may fail ocasionally due to misconfigured timeouts, this can specially
+   * hurt on archival chains where big operations require more time (and grater
```
Comment (nymd) [inline src/services/cherry-picker.ts]: * greater

File: src/services/cherry-picker.ts [hunk 5]
```diff
@@ -218,6 +235,57 @@ export class CherryPicker {
...
+   * timeouts) to operate. When a node continously fail due to the misconfiguration,
+   * is removed from the session
+   * @param blockchain
+   * @param serviceNode
+   * @param elapsedTime
+   * @param timeout
```
Comment (nymd) [inline src/services/cherry-picker.ts]: I think this should be timeoutCounter or something similar.

Comment (nymd) [reply src/services/cherry-picker.ts]: or even just `timeouts`

**Review by nymd — CHANGES_REQUESTED** [review]

File: src/application.ts [hunk 6]
```diff
@@ -77,7 +78,8 @@ export class PocketGatewayApplication extends BootMixin(ServiceMixin(RepositoryM
     const defaultLogLimitBlocks: number = parseInt(DEFAULT_LOG_LIMIT_BLOCKS) || 10000
     const aatPlan = AAT_PLAN || AatPlans.PREMIUM
     const redirects: string | object[] = REDIRECTS || ''
-    const commitHash: string | string = COMMIT_HASH || ''
+    const commitHash: string = COMMIT_HASH || ''
+    const archivalChains: string[] = (ARCHIVAL_CHAINS || []).split(',')
```
Comment (nymd) [inline src/application.ts]: Will this fail if someone accidentally puts a space after the commas?

**Review by nymd — APPROVED** [review]

File: src/application.ts (hunk 6 above)
Comment (rem1niscence) [reply src/application.ts]: not fail per se but will add the space to the result, will add a small fix for that

**Pull request merged**